subsys/bootloader/cmake/sign.cmake

```diff
@@ -4,61 +4,75 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
-set(sign_depends kernel_elf ${KEY_FILE_DEPENDS})
 
-set(hash_file firmware.sha256)
-set(signature_file firmware.signature)
-set(SIGNATURE_PUBLIC_KEY_FILE public.pem)
+#TODO: Detect what kernel_elf to depend on and resolve what
+#      image should be hashed and signed by b0.
+
+set(KERNEL_ELF_TO_BE_SIGNED kernel_elf)
```

Comment:
The idea here is to change this in the future to use a generator expression so that b0 knows what's in front of it and will sign the kernel/target hex which it is supposed to validate and boot. Look at the comment above.